docs/lifecycle/hooks/hooks.md

```diff
@@ -0,0 +1,347 @@
+
+In order to enable this reactivity feature, Vue takes advantage of the `Object.defineProperty` method introduced in ES5, using it to convert every property into a getter/setter.
+
+Below, is a very ilustrative diagram on how Vue handles reactivity:
+
+<img :src="$withBase('/reactivity.png')" style="text-align:center"/>
+
+## Data rendering
+
+On the early steps of this bootcamp, we learnt that the `template` is the suggested place to render our data (or the .html file when it comes to splitted components). However, there are specific cases where we want to use a more in depth approach using the `render` function. Let's take a look at this practical use case comming from the official documentation:
+
+```html
+<template>
+  <h1 v-if="level === 1">
+    <slot></slot>
+  </h1>
+  <h2 v-else-if="level === 2">
+    <slot></slot>
+  </h2>
+  <h3 v-else-if="level === 3">
+    <slot></slot>
+  </h3>
+  <h4 v-else-if="level === 4">
+    <slot></slot>
+  </h4>
+  <h5 v-else-if="level === 5">
+    <slot></slot>
+  </h5>
+  <h6 v-else-if="level === 6">
+    <slot></slot>
+  </h6>
+</template>
+```
+
+```javascript
+module.exports = {
+  props: {
+    level: {
+      type: Number,
+      required: true
+    }
+  }
+};
+```
+
+The first impression about this template is the imperative and verbose code, that makes it harder to mantain. Just think about the fact you want to add another subelement or behavior to each type of header, you'll fin yourself repeating code through all the possible if cases. On situations like this is where the `render` functions comes into place to create the difference and use a different approach, let's refactor this component using it.
+
+```javascript
+module.exports = {
+  render: createElement => {
+    return createElement(
+      "h" + this.level, // tag name
+      this.$slots.default // array of children
+    );
+  },
+  props: {
+    level: {
+      type: Number,
+      required: true
+    }
+  }
+};
+```
+
+The component now look much more simpler and easy to mantain, but we have a lot of new approaches here, let's brekdown the ideas:
+
+- The `render`function returning just one element, which is important due to component's definition. In order to do that, the `createElement` function is a key step. This function will be always received as a parameter.
+- The `this.$slots.default` is an important property in the Vue instance, basically stores a default slot tag.
+- Within the render funciton, we can still use the `this`context variable to access properties.
+
+## Directives and Data Binding
+
+This two concepts are essential when it comes to Vue.js. Let's first take a look at directives and then the way it's related to data binding:
```

Comment:
_This two concepts are essential_ I think here we should use _**These**_ as we're talking about plurals.